compiler/rustc_mir_transform/src/inline.rs

```diff
@@ -209,6 +209,9 @@ impl<'tcx> Inliner<'tcx> {
 
         let callee_body = try_instance_mir(self.tcx, callsite.callee.def)?;
         self.check_mir_body(callsite, callee_body, callee_attrs, cross_crate_inlinable)?;
+        if let Some(_) = callee_body.tainted_by_errors {
+            return Err("Body is tainted");
+        }
```

Comment:
Should this go at the beginning of check_mir_body ?

Reply:
Oh, hey, exactly the person who I was looking for. 

Yeah, I can do that